app/services/foreman/plugin/role_lock.rb

```diff
@@ -7,33 +7,39 @@ def initialize(plugin_id)
         @plugin_id = plugin_id
       end
 
-      def register_role(name, permissions, role_registry)
+      def register_role(name, permissions, role_registry, description = nil)
```

Comment:
Above you default description to `''`, so this will never be called without a description parameter afaict, and will override the role description with blank string if a description isn't passed from the plugin

Reply:
TBH it doesn't really matter to me if we default to `nil` or `''`, but in any case the default here will never be used since this function is only called from `Plugin#role`, so it is never called with just 3 parameters (except maybe in the tests).

Reply:
I don't really understand why it's a problem, whenever we call this method we need to explicitly send `''` as a description now. Yes, it's one place ATM (3 in tests). But anyway, I pushed the new version without description default value in here.